src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_CAPASITY = 10;
+    private Object[] elementData;
+    private int arrayCapasity;
+    private int ammountOfElements;
+
+    public ArrayList() {
+        arrayCapasity = DEFAULT_ARRAY_CAPASITY;
+        ammountOfElements = 0;
+        this.elementData = (T[]) new Object[DEFAULT_ARRAY_CAPASITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (ensureCapacity(ammountOfElements + 1)) {
+            elementData[ammountOfElements++] = value;
+        } else {
+            Object[] oldData = elementData;
+            newArrayCapasity(arrayCapasity);
+            System.arraycopy(oldData, 0, elementData, 0, ammountOfElements + 1);
+            elementData[ammountOfElements++] = value;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < ammountOfElements + 1 || index < 0) {
+            if (ensureCapacity(ammountOfElements + 1)) {
+                System.arraycopy(elementData, index, elementData,
+                        index + 1, ammountOfElements + 1 - index);
+                elementData[index] = value;
+                ammountOfElements++;
+            } else {
+                Object[] oldData = elementData;
+                newArrayCapasity(arrayCapasity);
+                System.arraycopy(oldData, index, elementData,
+                        index + 1, ammountOfElements + 1 - index);
+                elementData[index] = value;
+                ammountOfElements++;
+            }
+        } else {
```

Comment:
this else block is redundant, try to make 
if (check index if it fits) { 
     throw new Array....
}
and you won't need else then